File: csharp/GetOptionsPrices.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using Intrinio.SDK.Api;
using Intrinio.SDK.Client;
using Intrinio.SDK.Model;

namespace Example
{
  public class GetOptionsPricesExample
  {
    public static void Main()
    {
      Configuration.Default.AddApiKey("api_key", "YOUR_API_KEY");

      var optionsApi = new OptionsApi();
      string identifier = "AAPL190318C00300510";  // string | The Intrinio ID or code of the options contract to request prices for.
      string startDate = null;  // string | Return option contract prices on or after this date. (optional) 
      string endDate = null;  // string | Return option contract prices on or before this date. (optional) 
      decimal? pageSize = 100;  // decimal? | The number of results to return (optional)  (default to 100)
      string nextPage = null;  // string | Gets the next page of data from a previous API call (optional) 

      try
      {
        ApiResponseOptionPrices result = optionsApi.GetOptionsPrices(identifier, startDate, endDate, pageSize, nextPage);

        List<OptionPrice> prices = result.Prices;
        Option option = result.Option;

        Console.WriteLine("*** OPTION ***");
        Console.WriteLine("ID:         " + option.Id);
        Console.WriteLine("Code:       " + option.Code);
        Console.WriteLine("Ticker:     " + option.Ticker);
        Console.WriteLine("Expiration: " + option.Expiration);
        Console.WriteLine("Strike:     " + option.Strike);
        Console.WriteLine("Type:       " + option.Type);

        Console.WriteLine();
        Console.WriteLine(prices.Count + " options prices for " + identifier + "!");
        Console.WriteLine("-----------------------------------------------------------------------------------------------------");
        Console.WriteLine();

        prices.ForEach(delegate (OptionPrice price)
        {
          Console.WriteLine();
          Console.WriteLine("Date:                      " + price.Date);
          Console.WriteLine("Close:                     " + price.Close);
          Console.WriteLine("Close bid:                 " + price.CloseBid);
          Console.WriteLine("Close ask:                 " + price.CloseAsk);
          Console.WriteLine("Volume:                    " + price.Volume);
          Console.WriteLine("Volume bid:                " + price.VolumeBid);
          Console.WriteLine("Volume ask:                " + price.VolumeAsk);
          Console.WriteLine("Trades:                    " + price.Trades);
          Console.WriteLine("Open interest:             " + price.OpenInterest);
          Console.WriteLine("Open interest change:      " + price.OpenInterestChange);
          Console.WriteLine("Next day open interest:    " + price.NextDayOpenInterest);
          Console.WriteLine("Implied volatility:        " + price.ImpliedVolatility);
          Console.WriteLine("Implied volatility change: " + price.ImpliedVolatilityChange);
          Console.WriteLine("Delta:                     " + price.Delta);
        });
      }
      catch (Exception e)
      {
        Debug.Print("Exception when calling OptionsApi.GetOptionsPrices: " + e.Message );
      }
    }
  }
}

File: python/get_options_prices.py
from __future__ import print_function
import time
import intrinio_sdk
from intrinio_sdk.rest import ApiException
from pprint import pprint
 
intrinio_sdk.ApiClient().configuration.api_key['api_key'] = 'YOUR_API_KEY'
 
options_api = intrinio_sdk.OptionsApi()
 
identifier = 'AAPL190318C00300510' # str | The Intrinio ID or code of the options contract to request prices for.
 
try:
    api_response = options_api.get_options_prices(identifier)
    option = api_response.option
    prices = api_response.prices
    
    print("*** OPTION ***")
    print("ID:         %s" % (option.id))
    print("Code:       %s" % (option.code))
    print("Ticker:     %s" % (option.ticker))
    print("Expiration: %s" % (option.expiration))
    print("Strike:     %s" % (option.strike))
    print("Type:       %s" % (option.type))
    
    print()
    print("%s options prices for %s!" % (len(prices), identifier))
    print("------------------------------------------------------------")

    for price in prices:
        print()
        print("Date:                      %s" % (price.date))
        print("Close:                     %s" % (price.close))
        print("Close bid:                 %s" % (price.close_bid))
        print("Close ask:                 %s" % (price.close_ask))
        print("Volume:                    %s" % (price.volume))
        print("Volume bid:                %s" % (price.volume_bid))
        print("Volume ask:                %s" % (price.volume_ask))
        print("Trades:                    %s" % (price.trades))
        print("Open interest:             %s" % (price.open_interest))
        print("Open interest change:      %s" % (price.open_interest_change))
        print("Next day open interest:    %s" % (price.next_day_open_interest))
        print("Implied volatility:        %s" % (price.implied_volatility))
        print("Implied volatility change: %s" % (price.implied_volatility_change))
        print("Delta:                     %s" % (price.delta))
except ApiException as e:
    print("Exception when calling OptionsApi->get_options_prices: %s\n" % e)
